src/adjoint_common.jl

```diff
@@ -165,8 +165,20 @@ function adjointdiffcache(g,sensealg,discrete,sol,dg,f;quad=false,noiseterm=fals
         end
       end
     end
+
     if compile_tape(sensealg.autojacvec)
       paramjac_config = ReverseDiff.compile(tape)
+    elseif sensealg.autojacvec isa Bool && sensealg.autojacvec && DiffEqBase.isinplace(prob)
+      compile = try
+          !hasbranching(prob.f,copy(u0),u0,p,prob.tspan[2]))
```

Comment:
@simeonschaub `prob.f` mutates the first argument: is the copy required here for safety?

Reply:
Yes, the overdub will always run everything as well, so this is needed if you don't want the check to mutate `u0`. Might even need to be a `deepcopy` if elements of `u0` are allowed to be mutated as well.